src/main/java/com/conveyal/gtfs/loader/JdbcTableWriter.java

```diff
@@ -1497,36 +1553,43 @@ private void updateReferencingTables(
         for (Table referencingTable : referencingTables) {
             // Update/delete foreign references that have match the key value.
             String refTableName = String.join(".", namespace, referencingTable.name);
-            for (Field field : referencingTable.editorFields()) {
-                if (field.isForeignReference() && field.referenceTable.name.equals(table.name)) {
-                    // Get statement to update or delete entities that reference the key value.
-                    PreparedStatement updateStatement = getUpdateReferencesStatement(sqlMethod, refTableName, field, keyValue, newKeyValue);
-                    LOG.info(updateStatement.toString());
-                    int result = updateStatement.executeUpdate();
-                    if (result > 0) {
-                        // FIXME: is this where a delete hook should go? (E.g., CalendarController subclass would override
-                        //  deleteEntityHook).
-                        if (sqlMethod.equals(SqlMethod.DELETE)) {
-                            // Check for restrictions on delete.
-                            if (table.isCascadeDeleteRestricted()) {
-                                // The entity must not have any referencing entities in order to delete it.
-                                connection.rollback();
-                                String message = String.format(
-                                    "Cannot delete %s %s=%s. %d %s reference this %s.",
-                                    entityClass.getSimpleName(),
-                                    keyField.name,
-                                    keyValue,
-                                    result,
-                                    referencingTable.name,
-                                    entityClass.getSimpleName()
-                                );
-                                LOG.warn(message);
-                                throw new SQLException(message);
+            int result;
+            // Custom logic for calendar dates because schedule_exceptions does not reference calendar dates right now.
+            if (table.name.equals("schedule_exceptions") && referencingTable.name.equals("calendar_dates")) {
+                result = deleteCalendarDatesForException(id, namespace, table, refTableName);
+                LOG.info("Deleted {} entries in calendar dates associated with schedule exception {}", result, id);
+            } else { // General deletion
```

Comment:
nit. Move this comment into the else statement.

Reply:
Changed in https://github.com/conveyal/gtfs-lib/pull/376/commits/f035d18dc1721f03bfea4a1d5435184ab22f01e5